### PR TITLE
DEL-368: Upload downtime reports to s3

### PR DESCRIPTION
**Pull request by StanleyKaleta**
example cmds:
```
oc login ...
export DELOREAN_AWS_ACCESS_KEY_ID
export DELOREAN_AWS_SECRET_ACCESS_KEY
./delorean report query-report --config-file `pwd`/configurations/downtime-report-config.yaml -o `pwd`/downtime-report --s3-bucket=rhmi-test-data-storage-tf56 --pipeline-name="upgrade-54"
```

### CODE REVIEW

File: cmd/queryReport.go [hunk 1]
```diff
@@ -163,7 +190,7 @@ func init() {
 			}
 		},
 	}
-	pipelineCmd.AddCommand(cmd)
+	reportCmd.AddCommand(cmd)
```
Comment (wei-lee) [inline cmd/queryReport.go:193]: from the description of the commands, I think it's better to keep this command as a subcommand for `pipeline`. The reporting commands are mainly for importing data into Polarion or reportportal. Here we are still mainly about generating the report and upload to s3.

File: cmd/queryReport.go [hunk 2]
```diff
@@ -154,7 +167,21 @@ func init() {
 			if err != nil {
 				handleError(err)
 			}
-			c, err := newQueryReportCmd(kubeConfig, f)
+			awsKeyId, err := requireValue(AWSAccessKeyIDEnv)
+			if err != nil {
+				handleError(err)
+			}
+			awsSecretKey, err := requireValue(AWSSecretAccessKeyEnv)
```
Comment (wei-lee) [inline cmd/queryReport.go:174]: these keys should only be required if the `s3bucket` flag is set. It should be possible to run the command without uploading the report.

File: cmd/queryReport.go [hunk 3]
```diff
@@ -154,7 +167,21 @@ func init() {
...
+			if err != nil {
+				handleError(err)
+			}
+
+			s := session.Must(session.NewSession(&aws.Config{
+				Region:      aws.String(AWSDefaultRegion),
+				Credentials: credentials.NewStaticCredentials(awsKeyId, awsSecretKey, ""),
+			}))
```
Comment (wei-lee) [inline cmd/queryReport.go:182]: this should only be needed when it is required to upload the report.

File: cmd/queryReport.go [hunk 4]
```diff
@@ -246,6 +292,22 @@ func (c *queryReportCmd) run(ctx context.Context) error {
 		return err
 	}
 	fmt.Println("Report is generated:", outputFile)
+
+	zippedDir := c.outputDir + "/"
+	// zip the results
+	fmt.Println("Generating... " + zippedDir + c.zipFile)
+	err = utils.ZipFolder(zippedDir, zippedDir+c.zipFile)
```
Comment (wei-lee) [inline cmd/queryReport.go:299]: there is no need to zip the folder here. There is only one file that needs to be uploaded. If you want to zip the files here, make sure you will need to unzip them [here](https://github.com/integr8ly/delorean/blob/master/cmd/datahubImport.go#L161)

**Review by wei-lee — CHANGES_REQUESTED** [review]
see my comments above

File: cmd/queryReport.go (hunk 1 above)
Comment (StanleyKaleta) [reply cmd/queryReport.go:193]: ok

File: cmd/queryReport.go (hunk 4 above)
Comment (StanleyKaleta) [reply cmd/queryReport.go]: @wei-lee updated PR

**Review by wei-lee — APPROVED** [review]

**Pull request merged**